src/openrct2-ui/windows/EditorScenarioOptions.cpp

```diff
@@ -1121,6 +1134,29 @@ static void WindowEditorScenarioOptionsParkMousedown(rct_window* w, rct_widgetin
             }
             w->Invalidate();
             break;
+        case WIDX_FORBID_HIGH_CONSTRUCTION_SPINNER_INCREASE:
+            if (gConstructionHeightRestriction < 118)
+            {
+                gConstructionHeightRestriction++;
+            }
+            else
+            {
+                context_show_error(STR_CANT_INCREASE_FURTHER, STR_NONE, {});
+            }
+            w->Invalidate();
+            break;
+        case WIDX_FORBID_HIGH_CONSTRUCTION_SPINNER_DECREASE:
+            if (gConstructionHeightRestriction > 0)
```

Comment:
The minimum should probably be tree height, as forbidding less than that would be unreasonable (tree height already limits some taller flat rides). It definitely can't be 0.

```suggestion
            if (gConstructionHeightRestriction > TREE_HEIGHT)
```

Reply:
That's probably fair yeah, although with a lower limit you could enforce the player to build underground only?

Reply:
I've set the limit to 6m, which allows for stalls, several flat rides, and path, and even for tracked rides to go up once